src/rsvc/srv.c

```diff
@@ -1150,7 +1152,12 @@ ds_rsvc_start_aggregator(crt_rpc_t *source, crt_rpc_t *result, void *priv)
 
 	out_source = crt_reply_get(source);
 	out_result = crt_reply_get(result);
+	/* rc is error count, rc_errval an aggregation of values. */
 	out_result->sao_rc += out_source->sao_rc;
+
+	D_DEBUG(DB_MD, "\n");
+	out_result->sao_rc_errval = MIN(out_result->sao_rc_errval,
+					out_source->sao_rc_errval);
```

Comment:
What are the considerations behind MIN? Just want to understand it better.

Reply:
max would be problematic if out_result->sao_rc_errval = 0 because it would then ignore out_source->sao_rc_errval = -DER_NOSPACE (for example). Understand there are other aggregators that perform a test and use a pattern such as this:
  if (output_result->rc == 0) {
                 if (output_source->rc != 0)
                         output_result->rc = output_source->rc;
         }

Goal of this patch is to address at least these types of cases:
- all replicas experience the same error particularly -DER_NOSPACE
- all replicas experience either one particular error such as -DER_NOSPACE, or rc=0 (this scenario is actually borne out in some manual experiments I ran - e.g., 2 out of 3 server ranks got -DER_NOSPACE and the other rank succeeded).

But in the case that replicas experience more than one error value, the min function at least allows the code to choose one. Not sure if it intuitively makes sense, but use min combining to return the "most unusual" of the errors seen?